chain/chain/src/chain.rs

```diff
@@ -5865,22 +5862,66 @@ impl<'a> ChainUpdate<'a> {
         Ok(true)
     }
 
+    /// This function assumes `BlockInfo` is already retrievable from `epoch_manager`.
+    /// This can be achieved by calling `add_validator_proposals`.
+    #[cfg(feature = "new_epoch_sync")]
+    fn save_epoch_sync_info_if_finalised(&mut self, header: &BlockHeader) -> Result<(), Error> {
+        let block_info = self.epoch_manager.get_block_info(header.hash())?;
+        let epoch_first_block_hash = block_info.epoch_first_block();
+
+        if *epoch_first_block_hash == CryptoHash::default() {
+            // This is the genesis epoch. We don't have any fully finalised epoch yet.
+            return Ok(());
+        }
+
+        let epoch_first_block_info = self.epoch_manager.get_block_info(epoch_first_block_hash)?;
+        let prev_epoch_last_block_hash = epoch_first_block_info.prev_hash();
+
+        if *prev_epoch_last_block_hash == CryptoHash::default() {
+            // This is the genesis epoch. We don't have any fully finalised epoch yet.
```

Comment:
The prev epoch is genesis epoch, isn't it?

Reply:
If `epoch_first_block_info.prev_hash() == CryptoHash::default()`, then `epoch_first_block_info` id the genesis block, so this is the genesis epoch, and the previous epoch doesn't exist.